src/JwtFacade.php

```diff
@@ -4,23 +4,26 @@
 namespace Lcobucci\JWT;
 
 use Lcobucci\JWT\Encoding\JoseEncoder;
-use Lcobucci\JWT\Token as TokenInterface;
 use Lcobucci\JWT\Token\Parser;
 use Lcobucci\JWT\Validation\Constraint;
 use Lcobucci\JWT\Validation\SignedWith;
 use Lcobucci\JWT\Validation\ValidAt;
 use Lcobucci\JWT\Validation\Validator;
 
+use function assert;
+
 final class JwtFacade
 {
     public function parse(
         string $jwt,
         SignedWith $signedWith,
         ValidAt $validAt,
         Constraint ...$constraints
-    ): TokenInterface {
+    ): UnencryptedToken {
         $token = (new Parser(new JoseEncoder()))->parse($jwt);
 
+        assert($token instanceof UnencryptedToken);
```

Comment:
Ok, but I can't test the runtime exception since the returned value of `(new Parser)->parse()` is fixed.
Should I add the runtime check into infection ignore list too, so?

Reply:
What I would like to avoid is having to also modify this component when implementing nested tokens.
However, the alternative solution - relying on optional constructor injection of the interfaces instead of instantiating things on the method - would make things a bit more complicated now.

Let's just accept this trade-off for now...